ipatests/test_integration/test_epn.py

```diff
@@ -344,22 +349,114 @@ def test_EPN_smoketest_3(self):
         expected_users = ["user1", "user3", "user7", "user14", "user28"]
         assert sorted(user_lst) == sorted(expected_users)
 
-    def test_EPN_nbdays(self):
+    def test_EPN_nbdays_0(self, cleanupmail):
         """Test the to/from nbdays options (implies --dry-run)
 
            We have a set of users installed with varying expiration
            dates. Confirm that to/from nbdays finds them.
+
+           Make sure --dry-run does not accidentally send emails.
         """
 
-        # Compare the notify_ttls values
+        # Use the notify_ttls values with a 1-day sliding window
         for i in self.notify_ttls:
             user_list = []
-            (stdout_text_client, unused) = self._check_epn_output(
-                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True)
+            (stdout_text_client, unused, unused) = self._check_epn_output(
+                self.clients[0], from_nbdays=i, to_nbdays=i + 1, dry_run=True
+            )
             for user in json.loads(stdout_text_client):
                 user_list.append(user["uid"])
             assert len(user_list) == 1
-            assert user_list[0] == "user%d" % i
+            userid = "user{id}".format(id=i)
+            assert user_list[0] == userid
+
+            # Check that the user list is expected for any given notify_ttls.
+            (stdout_text_client, unused, unused) = self._check_epn_output(
+                self.clients[0], to_nbdays=i
+            )
+            user_list = [user["uid"] for user in json.loads(stdout_text_client)]
+            assert len(user_list) == 1
+            assert user_list[0] == "user{id}".format(id=i - 1)
+
+            # make sure no emails were sent
+            result = self.clients[0].run_command(['ls', '-lha', '/var/mail/'])
+            assert ".." in result.stdout_text
```

Comment:
This just asserts that there is a parent directory, what does that prove?

Reply:
Nothing, will fix, thanks for the catch!

Reply:
The ls will raise an exception if execution fails so I think you're safe without the assert.

Reply:
OK, I've removed it.